tests/roundtrip.rs

```diff
@@ -53,3 +53,50 @@ fn round_trip_from_image(original: &[u8]) {
         }
     }
 }
+
+#[test]
+fn encode_roundtrip_few_colors() {
+    const WIDTH: u16 = 128;
+    const HEIGHT: u16 = 128;
+
+    // Build an image with a single red pixel, that NeuQuant won't
+    // sample, in order to check that we do appropriatelyq specialise the
+    // few-colors case.
+    let mut pixels: Vec<u8> = vec![255; WIDTH as usize * HEIGHT as usize * 4];
+    // Top-left pixel is always sampled, so use the second pixel.
+    pixels[5] = 0;
+    pixels[6] = 0;
+    // Set speed to 30 to handily avoid sampling that one pixel.
+    //
+    // We clone "pixels", since the parameter is replaced with a
+    // paletted version, and later we want to compare the output with
+    // the original RGBA image.
+    let frames = [Frame::from_rgba_speed(WIDTH, HEIGHT, &mut pixels.clone(), 30)];
+
+    let mut buffer = vec![];
```

Comment:
I've usually seen this written as `let mut buffer = Vec::new()` but it doesn't really matter

Reply:
Some Rust team members swear on using the macro because it's closer to array syntax. Doesn't really matter to me either.

Reply:
This was C&P from `encode_roundtrip`, trying to be consistent. My uninformed preference is Vec::new(), but I'll leave as-is if no-one's too bothered!